### ISSUE TITLE
Start menu changes location when using search

### ISSUE THREAD

**Issue by evlo** [valinet/ExplorerPatcher#39]
when i open start and start typing it jumps from left corner where I have it set up to the center of the screen
https://www.youtube.com/watch?v=El3PD1XgzUw

**Comment by valinet** [comment]
It sometimes happens, especially when Explorer is restarted frequently. Just try a restart and it gets back to normal. Normally, it is fine. On a cold boot, it works just fine.

**Comment by kaivol** [comment]
I am also experiencing this bug. 
I have the taskbar moved to the right of my screen, so the start menu button is in the top right corner. When I open the startmenu it is correctly placed in the top right corner, but when I start typing the startmenu moves to the middle of the right side of the screen. 

`Location on screen` is set to `Left`.
This behaviour does not change after restarts.

**Comment by evlo** [comment]
For me the explorer restart did help, hopefully this will not be anoying :)

**Comment by valinet** [comment]
Has it happened again? Does it happen often? Without a million Explorer restarts, just using it normally, does it happen often?

Because personally I don't experience it at all. And I think it is normal for them to get messed up if Explorer is restarted a million times, themes are changed, whatever. But under normal operation, I think the behavior is consistent, with them showing at the right places.

Also, keep in mind that I don't think anyone at Microsoft tests these with the exotic taskbar placements, so there may be untested edge cases when they simply bug out. I don't know each and every detail, neither do I work at Microsoft or have access to the source code.

**Comment by valinet** [comment]
> In my case (taskbar on right side) the startmenu always behaves as described, also immediately after reboots. When the taskbar is position at the bottom it indeed works as intended.
> 
> If I understand correctly this is a limitation I the OS and not something this application can fix, right?

I don’t know. I have yet to fully understand how the code which tells Start where to position itself works. In typical Microsoft fashion, it is mot easy to understand, considering they just keep it for their own discretion only. The Start menu and search do not position themselves to left/center depending on the “taskbar alignment” setting in the main Settings app. Of course it is more complicated. To have them react:
* initial versions of the patcher hooked functions in StartMenuExpeienceHost.exe and SearchUx which were directly responsible with positioning the windows. This approach had a few problems, including having to download a huge symbol files
* recently, I disassembled StartIsBack and checked how it gets those to position properly. There is some interface that provides this info, and one has to intercept the call that requests this interface and answer with an in-house interface where the method which is used to determine the position is altered to our liking. Of course, NONE of this is documented, and StartIsBack is also not available in source form and also without symbols, so it is pretty hard to figure out what the interface structure is exactly and what everything in there does. For what is worth, the current implementation is an own port of the disassembly of StartIsBack, so it should behave the same. You can test StartIsBack in your scenario, if everything behaves the same, then it is some other kind of problem that also may be fixable, no doubt, while if with StartIsBack search gets positioned at a certain place all the time, so different behavior than what you describe with ExplorerPatcher, then it may be in the ported code

So yeah, idk, of course if I find a workaround, I will update the app with a fix, but things move slowly, there is no source code, it’s detective work 24/7.

**Comment by VanaKekchose** [comment]
Hello! I was wondering if there was a workaround found for this issue? I'm having the exact same problem (my task bar is to the right, I click on the windows icon everything is fine, but I start typing and it goes to the bottom left of the screen). Trying to respond to the questions you asked the other user:

"Has it happened again? Does it happen often?" It has happened everytime I typed so far (I've installed Explorer just an hour ago and have experienced this from the get go)

"Without a million Explorer restarts, just using it normally, does it happen often?" Yes, I tried to cold reboot it as you suggested, but it's still the same

Thank you for all the time you've put into this whole project, appart from this small issue, it's 100% worth using Explorer, it really is a game changer aha

(I'm new to GitHub, so apologies if I do something wrong, or write this in the wrong forum lol)
